src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,118 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.exeptions.DataProcessingException;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.DbConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        String query = "INSERT INTO products (product_name, product_price)"
+                + " VALUES (? , ?)";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, item.getName());
+            preparedStatement.setDouble(2, item.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet resultKey = preparedStatement.getGeneratedKeys();
+            if (resultKey.next()) {
+                item.setId(resultKey.getLong(1));
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Item has not been added/n" + e);
```

Comment:
```suggestion
            throw new DataProcessingException("Item has not been added", e);
```
do not concatenate message and exception object

Reply:
Done